src/main/java/org/betonquest/betonquest/compatibility/Compatibility.java

```diff
@@ -154,17 +154,17 @@ private void hook(final Plugin hook) {
         }
 
         // hook into the plugin if it's enabled in the config
-        if ("true".equalsIgnoreCase(plugin.getConfig().getString("hook." + name.toLowerCase(Locale.ROOT)))) {
+        if ("true".equalsIgnoreCase(betonQuest.getConfig().getString("hook." + name.toLowerCase(Locale.ROOT)))) {
             LOG.info(null, "Hooking into " + name);
 
             // log important information in case of an error
             try {
                 integrator.hook();
                 hooked.add(name);
-            } catch (final HookException exception) {
+            } catch (final Exception | LinkageError exception) {
```

Comment:
I think with that change we can also delete the HookException

Reply:
I don't think so, it is used by the unsupported version exception which is thrown by some intergrations if an invalid version is detected. 

It could also be usefull for an integration rework.

Reply:
@seyfahni what do you think about this? It is true, that the Invalid Version exception extends it, but there is no other usage so i think the Invalid Version can consume the code of the HookException.

Reply:
No, I would keep the HookException for hooks to be able to specify a non-version hooking failure.

Reply:
But i would change the line like this:
```suggestion
            } catch (final HookException | RuntimeException | LinkageError exception) {
```
This way we keep the contract that a hook can fail with a HookException but prevent uncaught RuntimeExceptions or LinkageErrors to ruin our day by disabling the plugin.

I would actually separate the handling of the hook exception and the others: HookExceptions are a known faulty state and should have a user-friendly message telling them how to solve it. So no stacktrace needed there.

RuntimeExceptions and LinkageErrors on the other hand are errors of the Integrator itself: Either due to it not being able to handle the latest version of the plugin or by not catching possible RuntimeExceptions by itself. So it has to be presented with stacktrace and everything.

Reply:
I really like that approach, it looks very clean.  

I think it's safe to say that all uncaught occurring Exceptions do extend RuntimeException.
Otherwise it should lead to a compiler error.